lib/modules/selectedEntry.js

```diff
@@ -167,12 +167,36 @@ function _selectedThing() {
 	return selectedThing;
 }
 
-const listeners = new $.Callbacks();
+const listeners = { beforeScroll: [], beforePaint: [], idle: [] };
 
-export function addListener(callback) {
-	listeners.add(callback);
+export function addListener(callback, when = 'idle') {
+	listeners[when].push(callback);
 }
 
+const runCallbacks = (() => {
+	function runListeners(listeners, newSelected, oldSelected, options) {
+		for (const listener of listeners) listener(newSelected, oldSelected, options);
+	}
+
+	let idleCallback;
+	const requestIdle = window.requestIdleCallback ?
+		window.requestIdleCallback : // XXX Only available in Chrome / Opera
+		fn => requestAnimationFrame(() => requestAnimationFrame(fn)); // Approximation for "Don't do anything this frame"
+	const runIdle = callback => {
```

Comment:
IIUC this ensures that the callback will only be called once if `runIdle` is called multiple times before an idle period.

If so, this abstraction can be exported from `async.js` as `idleThrottle` (just like `frameDebounce`, which should really be "throttle" rather than "debounce").

i.e. we would end up with

```js
const runIdle = idleThrottle(runListeners);
```